time/README.md

```diff
@@ -0,0 +1,25 @@
+# Moonbit/Core Time
+
+## Overview
+
+Package time provides functionality for measuring and manipulating time.
+
+The calendrical calculations always assume a ISO 8601 calendar, with no leap seconds.
+
+To create a datatime that represents the current time, you need to obtain the unix second and time zone offset from the [wasi](https://mooncakes.io/docs/#/peter-jerry-ye/wasi/) package (wasm-gc backend) or other FFI functions, and manually create a datetime.
+
+```
+// creates a UTC+8 fixed time zone.
+let zone = @time.fixed_zone("Asia/Shanghai", 8*60*60)
```

Comment:
There is a database containing all the standard code and offset: https://www.iana.org/time-zones

I start to hesitate whether we should make it part of the core or maintain it as a separate package

Reply:
The time-zone implementation only supports fixed offset now. Maybe we can refer to how Python integrates time zone data into the standard library ([PEP 615](https://peps.python.org/pep-0615/)).

Reply:
Considering we are a compiled language with multiple backends, and one of the backend (wasm) has multiple runtime (web or wasi), I'm not sure if we can really refer to Python's solution.

Reply:
IMO, we can add public `TimeZone` and `TimeZoneOffset` traits and make zone-related codes depend on these traits. We have two options for the time zone database:
- **Maintaining an independent `tzdata` package.** Only retain a `FixedZone` in the core library. If users need an unfixed time zone, they can use the `tzdata` package to construct a `TimeZone` instance with offset transition rules from the IANA database.
- **Embedding `tzdata` in the core library.** We need to generate moonbit codes from the IANA database to hold the time zone data and implement a lookup mechanism for constructing the `TimeZone` instance. We need to bump the version of the core library when the IANA database updates.